### ISSUE TITLE
Missing Test Case - 323. Number of Connected Components in an Undirected Graph

### ISSUE THREAD

**Issue by RyanShinra** [LeetCode-Feedback/LeetCode-Feedback#23897]
### LeetCode Username

RyanShinra

### Problem Number, Title, and Link

323 Number of Connected Components in an Undirected Graph https://leetcode.com/problems/number-of-connected-components-in-an-undirected-graph/description/?source=submission-noac

### Bug Category

Incorrect test case *(Output of test case is incorrect as per the problem statement)*

### Bug Description

Test case: n = 4, edges = [[2,3],[1,2],[1,3]]

Output
1
Expected
2

The 3 nodes form a triangle. There is only one piece of geometry.

### Language Used for Code

C++

### Code used for Submit/Run operation

```bash
class Solution {
public:
    set<int> visited;
    queue<int> toVisit;
    void doBfs(vector<vector<int>>& adj, vector<int>& inDeg, int start) {
        toVisit.push(start);
        visited.insert(start);
        while(toVisit.size() > 0) {
            int cur = toVisit.front();
            toVisit.pop();
            // Check all the neighbors
            for (int nbr = 0; nbr < adj[cur].size(); ++nbr) {
                if (adj[cur][nbr] == 1) {
                    if(!visited.contains(nbr)){
                        toVisit.push(nbr);
                        visited.insert(nbr);
                    }
                    adj[cur][nbr] = 0;
                    adj[nbr][cur] = 0;
                    inDeg[nbr] -= 1;
                    inDeg[cur] -= 1;
                }
            }
        }
        visited.clear();
    }
    int countComponents(int n, vector<vector<int>>& edges) {
        vector<vector<int>> adj(n, vector(n,0)); 
        vector<int> inDeg(n, 0);

        for(int i = 0; i < edges.size(); ++i) {
            const int a = edges[i][0];
            const int b = edges[i][1];
            adj[a][b] = 1;
            adj[b][a] = 1;
            inDeg[a] += 1;
            inDeg[b] += 1;
        }

        int pieceCount = 0;        

        for (int i = 0; i < inDeg.size(); ++i) {
            if (inDeg[i] > 0) {
                // the in degree will be set to zero by the time a traversal is finished
                ++pieceCount;
                doBfs(adj, inDeg, i);
            }
        }
        return pieceCount;
    }
};
```

### Expected behavior

The test case expected 2, but there is only 1 piece in the graph.

**Comment by 0IHI0** [comment]
try this
https://gofile.io/d/yMHabA
select the desired compiler in the installer

**Comment by RyanShinra** [comment]
Thanks for such a detailed reply; I'm sure you see lots of issues. There was some ambiguity with the problem description, but I read the comments and found that the unconnected nodes was an edge case I overlooked. Once I addressed that, I pretty much had it in the bag. The object lesson here being that I should look for such ambiguities and ask clarifying questions on the interview. 
Thanks again, this really was above and beyond what I was expecting.
